README.md

```diff
@@ -35,14 +35,17 @@ callback methods.
 ```elixir
 defmodule SlackRtm do
   use Slack
+  alias Slack.Message
 
   def handle_connect(slack, state) do
     IO.puts "Connected as #{slack.me.name}"
     {:ok, state}
   end
 
   def handle_event(message = %{type: "message"}, slack, state) do
-    send_message("I got a message!", message.channel, slack)
+    msg = Message.new_from_event(slack, message)
```

Comment:
I'm not sure this makes a lot of sense.

Is there a reason we wouldn't pass a message struct to `handle_event` directly?

Reply:
Maybe. There are a lot of types of events and most of them probably won't map nicely onto entities in the domain.

Reply:
I lean toward just using maps for events. Mostly because i'm lazy and don't really want to implement a wrapper for [every event type](https://api.slack.com/events). 

However, events relate to domain entities like users, channels, messages, etc. Even if events are modeled as maps it will be convenient to be able to build representations of the related domain entities from events. Conceptually, there is a big difference between the event notifying us that a thing was (for example) created and the thing itself. Consider the [`channel_created`](https://api.slack.com/events/channel_created) event vs the [conversation/channel](https://api.slack.com/types/conversation) entity.

Reply:
I think that maps are fine, but a user should never have to create a `Message`, `Channel`, or `User` struct manually. The library should handle that for the user.

In both of those examples I think a map is fine as long as the channel in that map is a `Channel` struct. If we don't do that we risk creating a confusing inconsistency in the library.

Reply:
:+1: I'm on the same page. At least as a long-term goal.

The downside is that change feels all or nothing-ish to me. Changing (for example) the `channel` member of the `message` event from a channel id to a `Channel` struct is a breaking change. Which means we'd want to those changes to land all at once to limit the damage.

I'd rather leave the bot interface as is until we have time to build up the domain entity abstractions. Once we have highly capable domain entities we could revamp the bot interface in one fell swoop. The in-between time is a little awkward, but it is not terrible. Particularly, if we provide convenience functions extracting domain entities from the most common event types.

Reply:
I definitely see it as an all or nothing change.

I don't think the awkwardness is worth it here since there's no real benefit yet. I think another challenge that hasn't been brought up yet is that we support both the web api and the RTM api. This change blurs that line quite a bit and I'm not sure if that is or isn't the correct abstraction yet.